config/routes.rb

```diff
@@ -15,7 +15,9 @@
   devise_for :users, controllers: { registrations: 'users/registrations',
                                     omniauth_callbacks:
                                     'users/omniauth_callbacks' }
-  resources :calculators, only: %i[index show], param: :slug
+  resources :calculators, only: %i[index show], param: :slug do
+    post :calculate, on: :member
```

Comment:
More straightforward way to cover in block
```suggestion
  member do
    post :calculate
  end
```